src/main/kotlin/org/rust/ide/hints/RsInlayTypeHintsProvider.kt

```diff
@@ -126,6 +130,20 @@ class RsInlayTypeHintsProvider : InlayHintsProvider<RsInlayTypeHintsProvider.Set
                 }
             }
 
+            private fun presentTypeArguments(declaration: RsLetDecl) {
+                val fullType = declaration.pat?.childOfType<RsPatBinding>()?.type ?: return
```

Comment:
This code does not work in case of complex patterns like these:
```Rust
let (xs, ys): (Vec<_>, Vec<_>) = (vec![1,2,3], vec![true, false]);
let S { x, y }: S<_> = S { x: 1, y: 2 };
```

Probably, all you have to do to support such cases is replace this line with
```Kotlin
val fullType = declaration.pat?.type ?: return
```

And please don't forget to add such tests

Reply:
Hi, just as a note on the side. This statement:
```Rust
val fullType = declaration.pat?.type ?: return
```
does not fully solve the issue.
the above stated example doesn't seem to resolve the types correctly.

the statement:
```Rust
let (xs, ys): (Vec<_>, Vec<_>) = (vec![1,2,3], vec![true, false]);
```
evaluates to:
<img width="576" alt="Screenshot 2020-04-18 at 12 22 02" src="https://user-images.githubusercontent.com/22869919/79635111-422fc000-816f-11ea-8064-a6f5a4926d56.png">